packages/compile-solidity/compilerSupplier/loadingStrategies/VersionRange.js

```diff
@@ -104,7 +104,7 @@ class VersionRange extends LoadingStrategy {
       attemptNumber: index + 1
     });
     try {
-      const response = await request.get(url);
+      const response = await request.get(url, { gzip: true, timeout: 30000 });
```

Comment:
Can we make this configurable? Something like:


```javascript
module.exports = {
  compilers: {
    solc: {
      obtain: {
        gzip: true,
        timeout: 30000
      }

      /* ... rest of solc config ... */
    }

    /* ... other compiler configs ... */
  }

  /* ... rest of config ... */
}
```

I used `obtain` as a name to match the `truffle obtain` command, but would love other ideas for naming/structure.

Reply:
I think It‘s not necessary to make this configurable.

1. Just like a browser, `gzip` and `timeout` are enabled by default.
2. `module.exports` will not affect `npm install -g truffle`(`postinstall` script will download `solc`)
3. `solc` is a `js`, `gzip` will reduce download size from `12MB` to `2MB`

Reply:
Alright you've convinced me!